### PR TITLE
THEEDGE-2293 refactor image update to event-driven 

### PR DESCRIPTION
**Pull request by loadtheaccumulator**
Signed-off-by: Jonathan Holloway <jholloway@redhat.com>

# Description

Refactor image update to EDA

Fixes # (issue) THEEDGE-2293

## Type of change

What is it?

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation update
- [ ] Tests update
- [x] Refactor

# Checklist:

- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I run `make pre-commit` to check fmt/vet/lint/test-no-fdo

### CODE REVIEW

File: pkg/models/events.go [hunk 1]
```diff
@@ -263,5 +263,5 @@ type EdgeImageRequestedEventPayload struct {
 type EdgeImageUpdateRequestedEventPayload struct {
 	EdgeBasePayload
 	NewImage Image `json:"new_image"`
-	OldImage Image `json:"old_image"`
+	//OldImage Image `json:"old_image"`
```
Comment (loadtheaccumulator) [inline pkg/models/events.go:266]: @AaronH88 see my comment in pkg/routes/images.go below.

File: pkg/services/image/events.go [hunk 2]
```diff
@@ -1,15 +1,11 @@
...
-	"github.com/redhatinsights/edge-api/pkg/models"
-)
-
 // EdgeMgmtImageEventInterface is the interface for the image microservice(s)
 type EdgeMgmtImageEventInterface interface {
 	Consume() error // handles the execution of code against the data in the event
 }
 
-// RegisteredEvents maps the record key to a corresponding event struct for handling
+/* // RegisteredEvents maps the record key to a corresponding event struct for handling
```
Comment (loadtheaccumulator) [inline pkg/services/image/events.go:8]: I can delete this commented section now. This is part of a future refactor to allow for a more simple consumer loop.

File: pkg/services/images.go [hunk 3]
```diff
@@ -355,9 +355,9 @@ func (s *ImageService) UpdateImage(image *models.Image, previousImage *models.Im
 
 	s.log = s.log.WithFields(log.Fields{"updatedImageID": image.ID, "updatedCommitID": image.Commit.ID})
 
-	s.log.Info("Image Updated successfully - starting bulding process")
+	s.log.Info("Image Updated successfully - starting building process")
 
-	go s.postProcessImage(image.ID)
+	//	go s.postProcessImage(image.ID)
```
Comment (loadtheaccumulator) [inline pkg/services/images.go:360]: Delete this.

**Comment by AaronH88** [comment]
overall it looks good.

What do you think of putting the marshal and unmarshal technique into a helper function of some kind?
This will be called a lot throughout our system and could be useful.

**Review by acosferreira — COMMENTED** [review]
lgtm

**Comment by loadtheaccumulator** [comment]
> What do you think of putting the marshal and unmarshal technique into a helper function of some kind? This will be called a lot throughout our system and could be useful.

@AaronH88 I whacked the whole thing and used a ridiculous, but effective, shadowing technique with a dummy struct that embeds the struct that shadows Data. Structs + interfaces = ok. Structs + interfaces + json = nightmare of all nightmares.

**Review by AaronH88 — APPROVED** [review]
/lgtm

**Pull request merged**